src/ddl_utils.c

```diff
@@ -0,0 +1,72 @@
+#include <postgres.h>
+#include <tcop/deparse_utility.h>
+
+/*
+ * Test if ddl command as an alter table add column
+ */
+PG_FUNCTION_INFO_V1(ddl_is_add_column);
+Datum
+ddl_is_add_column(PG_FUNCTION_ARGS)
+{
+	CollectedCommand *cmd = (CollectedCommand *) PG_GETARG_POINTER(0);
+	ListCell   *cell;
+	bool		is_add_column = false;
+
+	if (cmd->type == SCT_AlterTable)
+	{
+		foreach(cell, cmd->d.alterTable.subcmds)
+		{
+			CollectedATSubcmd *sub = lfirst(cell);
+			AlterTableCmd *subcmd = (AlterTableCmd *) sub->parsetree;
+
+			Assert(IsA(subcmd, AlterTableCmd));
+
+			switch (subcmd->subtype)
+			{
+				case AT_AddColumn:
+				case AT_AddColumnRecurse:
+					is_add_column = true;
+					break;
+				default:
+					break;
+			}
+		}
+	}
+
+	PG_RETURN_BOOL(is_add_column);
+}
+
+/*
+ * Test if ddl command as an alter table drop column
+ */
+PG_FUNCTION_INFO_V1(ddl_is_drop_column);
+Datum
+ddl_is_drop_column(PG_FUNCTION_ARGS)
```

Comment:
This function is nearly identical to `ddl_is_add_column`. I suggest moving the common code into its own C-function that returns, e.g., an enum (DDL_DROP_COLUMN, DDL_ALTER_COLUMN, DDL_OTHER) or the existing subtype. You can then call this function from both `ddl_is_drop_column` or `ddl_is_alter_column`.